Additional main.py commands

sjrct opened the pull request:
The `fetch` command allows you fetch a package from a remote repository into your local packages directory.

The `add-module` and `add-agent` commands allow you to add a module or agent to the `config.json` interactively. Options for the added module or agent are specified with the `options` field on the module or agent class.

sjrct committed:
Implement 'fetch' command in main.py

Fetch allows you fetch a package from a remote repository into your
packages directory.

sjrct committed:
Add 'add-module' and 'add-agent' commands

Allow a module or agent to be added to the configuration interactively.

Additional options per-agent or per-module can be specified with the
'options' field on he module or agent.

richteer commented on requirements.txt:
```diff
@@ -0,0 +1 @@
+GitPython
```
Can we make this an optional requirement? We should be able to conditionalize support if the module exists or not

sjrct replied:
Done

richteer commented on main.py:
```diff
@@ -69,10 +72,47 @@ def h_run(*args):
 	bot = halibot.Halibot()
 	bot.start(block=True)
 
+def h_fetch(*args):
```
This might serve better if it was pluggable, supporting multiple types of fetching. (Maybe shell out to wget, for example?)

sjrct replied:
Git repos, zip archives, and tarballs are now supported.

sjrct committed:
Pluggable repositories for fetch

zip and tar archives are now supported and differentiated from git repos
via the content of the repo url.
